x-pack/filebeat/input/s3/s3_integration_test.go

```diff
@@ -0,0 +1,417 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package s3
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/sqsiface"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/tests/resources"
+	awscommon "github.com/elastic/beats/x-pack/libbeat/common/aws"
+)
+
+const (
+	fileName          = "sample1.txt"
+	fileDir           = "./ftest/" + fileName
+	visibilityTimeout = 300 * time.Second
+)
+
+// GetConfigForTest function gets aws credentials for integration tests.
+func getConfigForTest() (config, string) {
+	awsConfig := awscommon.ConfigAWS{}
+	info := ""
+	queueURL := os.Getenv("QUEUE_URL")
+	if queueURL == "" {
+		info = "Skipping: $QUEUE_URL is not set in environment."
+	}
+
+	profileName := os.Getenv("AWS_PROFILE_NAME")
+	if profileName != "" {
+		awsConfig.ProfileName = profileName
+		config := config{
+			QueueURL:          queueURL,
+			AwsConfig:         awsConfig,
+			VisibilityTimeout: visibilityTimeout,
+		}
+		return config, info
+	}
+
+	accessKeyID := os.Getenv("AWS_ACCESS_KEY_ID")
+	secretAccessKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
+	sessionToken := os.Getenv("AWS_SESSION_TOKEN")
+	defaultRegion := os.Getenv("AWS_REGION")
+	if defaultRegion == "" {
+		defaultRegion = "us-west-1"
+	}
+
+	if accessKeyID == "" {
+		info = "Skipping: $AWS_ACCESS_KEY_ID or $AWS_PROFILE_NAME not set or set to empty"
+	} else if secretAccessKey == "" {
+		info = "Skipping: $AWS_SECRET_ACCESS_KEY not set or set to empty"
+	} else {
+		awsConfig = awscommon.ConfigAWS{
+			AccessKeyID:     accessKeyID,
+			SecretAccessKey: secretAccessKey,
+		}
+
+		if sessionToken != "" {
+			awsConfig.SessionToken = sessionToken
+		}
+	}
+	config := config{
+		QueueURL:          queueURL,
+		AwsConfig:         awsConfig,
+		VisibilityTimeout: visibilityTimeout,
+	}
+	return config, info
+}
+
+func testSetup(t *testing.T) context.CancelFunc {
+	t.Helper()
+	_, cancel := context.WithCancel(context.Background())
+	return cancel
+}
+
+func uploadSampleLogFile(t *testing.T, bucketName string, svcS3 s3iface.ClientAPI) {
```

Comment:
It could be good to mark this function as helper with `t.Helper()`. Consider it also for other helper methods here.